### PR TITLE
Add Arion gateway

### PR DESCRIPTION
**Pull request by DavidLiu506**
1 Send Arion gateway ip and mac to ACA.
2 Add consistent hash

### CODE REVIEW

File: schema/proto3/neighbor.proto [hunk 1]
```diff
@@ -43,6 +43,9 @@ message NeighborConfiguration {
         NeighborType neighbor_type = 1;
         string subnet_id = 2;
         string ip_address = 3;
+        string arion_group = 4;
```
Comment (xieus) [inline schema/proto3/neighbor.proto:46]: @DavidLiu506 Let us discuss this. Hopefully we could reuse Alcor basic data structure to support Arion.

File: schema/proto3/gateway.proto [hunk 2]
```diff
@@ -29,6 +29,7 @@ enum GatewayType {
     TGW = 2;    // Transit Gateway
     IGW = 3;    // Internet Gateway
     NGW = 4;    // NAT Gateway
+    ARION=5;
```
Comment (xieus) [inline schema/proto3/gateway.proto:32]: "ARION = 5"

File: schema/proto3/gateway.proto [hunk 3]
```diff
@@ -50,8 +51,17 @@ message GatewayConfiguration {
         uint32 port_inband_operation = 1;
     }
 
+    message arion {
+            string vpc_id = 1;
+            uint32 vni = 2;
+            string subnet_id = 3;
+            // port for in-band (same NIC channel) operation
+            uint32 port_inband_operation = 4;
```
Comment (xieus) [inline schema/proto3/gateway.proto:59]: Is this a port id or?

File: schema/proto3/arionmaster.proto [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+option java_package = "com.futurewei.alcor.schema";
+
+import "common.proto";
+import "goalstateprovisioner.proto";
+import "neighbor.proto";
+
+
+
+service ArionMasterService {
```
Comment (xieus) [inline schema/proto3/arionmaster.proto:29]: Is this the API provided by Arion Master? if so, we may put it in the Arion project and submodule from here.

File: schema/proto3/gateway.proto (hunk 3 above)
Comment (zzxgzgz) [reply schema/proto3/gateway.proto:59]: This is the port number for the in-band operation. Say, `port_inband_operation=1234`, then when ACA receives this Gateway State, it creates a UDP server at this port(1234) to receive the OAM packet sent by an Arion Wing.

File: schema/proto3/arionmaster.proto (hunk 4 above)
Comment (DavidLiu506) [reply schema/proto3/arionmaster.proto:29]: Good suggestions. Thank you!

File: schema/proto3/gateway.proto (hunk 3 above)
Comment (DavidLiu506) [reply schema/proto3/gateway.proto:59]: @zzxgzgz thank you!

Comment (xieus) [reply schema/proto3/gateway.proto:59]: @zzxgzgz based on the new design, the OAM packet is sent by a local OAM agent, not the Arion Wing, right?

File: services/data_plane_manager/src/main/java/com/futurewei/alcor/dataplane/entity/ArionWing.java [hunk 5]
```diff
@@ -0,0 +1,73 @@
+package com.futurewei.alcor.dataplane.entity;
+
```
Comment (xieus) [inline services/data_plane_manager/src/main/java/com/futurewei/alcor/dataplane/entity/ArionWing.java:2]: License statement..

File: services/data_plane_manager/src/main/java/com/futurewei/alcor/dataplane/cache/ArionWingCache.java [hunk 6]
```diff
@@ -33,7 +33,10 @@ free of charge, to any person obtaining a copy of this software and associated d
 @ComponentScan(value="com.futurewei.alcor.common.db")
 public class ArionWingCache {
 
+    // arionWingCache store Arion wing meta data. key is Arion Wing hash code and value is Arion wing meta data.
     private ICache<String, ArionWing> arionWingCache;
+
+    // arionWingGroupCache store Arion group meta data. key is Arion wing group name, value is Arion wing meta data.
```
Comment (xieus) [inline services/data_plane_manager/src/main/java/com/futurewei/alcor/dataplane/cache/ArionWingCache.java:39]: Is the value here Arion group info? What is the difference from ArionWing which is the wing meta data.

Comment (DavidLiu506) [reply services/data_plane_manager/src/main/java/com/futurewei/alcor/dataplane/cache/ArionWingCache.java:39]: There might be several Arion Wing belong to one group

**Review by xieus — APPROVED** [review]
LGTM

**Pull request merged**